src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = createBucketArray(DEFAULT_CAPACITY);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Node<K, V>[] createBucketArray(int capacity) {
+        return (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeeded();
+        int index = determineIndex(key);
```

Comment:
The `determineIndex` method uses `key.hashCode() & (buckets.length - 1)`, which may not handle negative hash codes correctly. Consider using `Math.abs(key.hashCode()) & (buckets.length - 1)` to ensure the index is always non-negative.